app/controllers/posts_controller.rb

```diff
@@ -26,6 +26,12 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @post.destroy
+    redirect_to user_posts_path(current_user), notice: 'Post Deleted Successfully'
+  end
+
   private
 
   def post_params
```

Comment:
Kindly add the `load_and_authorize_resource` to help use cancancan for post authorization in the Post controller.  The load_and_authorize_resource is a feature provided by the CanCanCan gem in Ruby on Rails. It simplifies the process of loading and authorizing resources in your controllers. 
With load_and_authorize_resource in place, CanCanCan will automatically load the appropriate resource based on the controller and action, and authorize the user's access to it based on the defined abilities in ability.rb. This helps you enforce authorization rules and reduces the amount of manual authorization code you need to write in your controllers.